lib/manageiq/rpm_build/generate_gemset.rb

```diff
@@ -96,6 +98,43 @@ def scrub
         cleanse_gemset
       end
 
+      def generate_dependency_manifest
+        where_am_i
+
+        FileUtils.rm_rf(manifest_dir) if manifest_dir.exist?
+        FileUtils.mkdir(manifest_dir)
+
+        shell_cmd("gem install license_finder")
+        generate_gem_manifest
+        generate_npm_manifest
+      end
+
+      def generate_gem_manifest
+        where_am_i
+        run_license_finder(miq_dir, "gem")
+      end
+
+      def generate_npm_manifest
+        where_am_i
+
+        cmd = "rake update:print_engines | grep path: | cut -d: -f2"
+        repos = AwesomeSpawn.run!(cmd, :chdir => miq_dir).output.split
+
+        # license_finder tries to look for all supported package manager, move out Gemfile
+        repo_with_gemfile = repos.select { |repo| File.exist?("#{repo}/Gemfile") }
+        repo_with_gemfile.each { |repo| FileUtils.mv("#{repo}/Gemfile", "#{repo}/Gemfile.save") }
+        run_license_finder(repos.join(" "), "npm")
+        repo_with_gemfile.each { |repo| FileUtils.mv("#{repo}/Gemfile.save", "#{repo}/Gemfile") }
```

Comment:
Minor, but I think this would read clearer as:

```ruby
# license_finder tries to look for all supported package managers, so temporarily remove the Gemfile
without_gemfiles(repos) do
  run_license_finder(repos.join(" "), "npm")
end
```

```ruby
def without_gemfiles(repos)
  repos = repos.select { |repo| File.exist?("#{repo}/Gemfile") }
  repos.each { |repo| FileUtils.mv("#{repo}/Gemfile", "#{repo}/Gemfile.save") }
  yield
ensure
  repos.each { |repo| FileUtils.mv("#{repo}/Gemfile.save", "#{repo}/Gemfile") }
end
```